src/dbnode/storage/types.go

```diff
@@ -502,6 +502,9 @@ type Shard interface {
 	// OpenStreamingDataReader creates and opens a streaming fs.DataFileSetReader
 	// on the latest volume of the given block.
 	OpenStreamingReader(blockStart xtime.UnixNano) (fs.DataFileSetReader, error)
+
+	// TryRetrieveWritableSeries attempts to retrieve a writable series.
```

Comment:
Can we explicitly mention that callers must call `entry.DecrementReaderWriterCount` after calling this method?

In addition to be honest would be to rename this to `TryRetrieveSeriesAndIncrementReaderWriterCount` - that way it's very clear what the contract is, even if you don't read the doc comment (the naming of method might also more likely make you read the comment too since you want to understand why it's incrementing the readerwriter count).